Remotion/Web/UnitTests/Core/UI/Controls/ListMenuImplementation/Rendering/ListMenuRendererTest.cs

```diff
@@ -243,6 +243,52 @@ public void RenderDiagnosticMetadataAttributes ()
       wrapper.AssertAttributeValueEquals(DiagnosticMetadataAttributes.IsDisabled, (!_control.Object.Enabled).ToString().ToLower());
     }
 
+    [Test]
+    public void Render_WithNoHeading_NoHeaderAndNoLabelledBy ()
+    {
+      SetUpGetPostBackLinkExpectations(true);
+
+      _control.Setup(e => e.Heading).Returns(WebString.Empty);
+      var wrapper = GetAssertedWrapper();
+
+      wrapper.AssertNoAttribute("aria-labelledby");
+      wrapper.AssertChildElementCount(1);
+    }
+
+    [Test]
+    public void Render_WithHeadingButNoHeadingLevel_RendersAndLinksSpan ()
+    {
+      SetUpGetPostBackLinkExpectations(true);
+
+      _control.Setup(e => e.Heading).Returns(WebString.CreateFromText("My heading"));
+      _control.Setup(e => e.HeadingLevel).Returns((HeadingLevel?)null);
+      var wrapper = GetAssertedWrapper();
+
+      wrapper.AssertAttributeValueEquals("aria-labelledby", "MyListMenu_Heading");
+
+      var span = wrapper.GetAssertedChildElement("span", 0);
+      span.AssertAttributeValueEquals("id", "MyListMenu_Heading");
+      span.AssertAttributeValueEquals("class", "screenReaderText");
```

Comment:
Here got one more bit: when it's a span, please add `hidden` attribute instead of css class `screenreadertext`. The text shouldn't be read just by virtually browsing the screen, it should only show up when reading the region.

Note : I am pretty sure that hidden is correct here. We do that with labels for controls, too, e.g. in bocdatetimevalue. Please verify

Reply:
If I read the code correctly, you got the hidden attribute for both span and  heading instead of just the heading. The heading as 'hidden' doesn't have value, please update to only do the span hidden and the heading remains screen reader.

Reply:
My bad I misread your comment. So if there is a heading level specified it should be targetable using the arrow keys and be the focus of the region? Because that is the current behavior and it seems kind of strange to target the invisible heading. Looks like this:
![image](https://github.com/re-motion/Framework/assets/5082673/28adaefa-ab49-41a9-b662-27bc35ff9609)